### PR TITLE
recursively remove pretrained step for testing

### PR DESCRIPTION
**Pull request by dreamerlin**

### CODE REVIEW

File: tools/test.py [hunk 1]
```diff
@@ -174,6 +174,16 @@ def main():
                               **cfg.data.get('test_dataloader', {}))
     data_loader = build_dataloader(dataset, **dataloader_setting)
 
+    if cfg.model.get('backbone', None):
+        # remove pretrain steps for testing
+        cfg.model.backbone.pretrained = None
+        cfg.model.backbone.pretrained2d = False
```
Comment (congee524) [inline tools/test.py:180]: maybe some `model.backbone` has no pretrained2d. such as C3D?

Comment (dreamerlin) [reply tools/test.py:180]: It doesn't matter, right?

Comment (dreamerlin) [reply tools/test.py:180]: Oops

File: tools/test.py [hunk 2]
```diff
@@ -174,6 +174,21 @@ def main():
                               **cfg.data.get('test_dataloader', {}))
     data_loader = build_dataloader(dataset, **dataloader_setting)
 
+    if cfg.model.get('backbone', None):
+        # remove pretrain steps for testing
+        if hasattr(cfg.model.backbone, 'pretrained'):
+            cfg.model.backbone.pretrained = None
+        if hasattr(cfg.model.backbone, 'pretrained2d'):
```
Comment (kennymckormick) [inline tools/test.py:181]: No need to set pretrained2d, since pretrained is None

File: tools/test.py [hunk 3]
```diff
@@ -174,6 +174,21 @@ def main():
...
+            cfg.model.backbone.pretrained2d = False
+
+        # For slowfast case
+        if cfg.model.backbone.get('slow_pathway', None):
```
Comment (kennymckormick) [inline tools/test.py:185]: We can recursively find all `pretrained` in the model config. There's no need to write a specific case for SlowFast

File: tools/test.py (hunk 2 above)
Comment (dreamerlin) [reply tools/test.py:181]: ur right

File: tools/test.py [hunk 4]
```diff
@@ -94,6 +94,19 @@ def parse_args():
...
 
+def turn_off_pretrained(cfg):
+    # recursively find all pretrained in the model config,
+    # and set them None to avoid redundant pretrain steps for testing
+    if hasattr(cfg, 'pretrained'):
+        cfg.pretrained = None
+
+    # recursively turn off pretrained value
+    for cfg_key in cfg:
+        sub_cfg = getattr(cfg, cfg_key)
```
Comment (congee524) [inline tools/test.py:105]: for _, sub_cfg in cfg.items():

File: tools/test.py [hunk 5]
```diff
@@ -94,6 +94,19 @@ def parse_args():
     return args
 
 
+def turn_off_pretrained(cfg):
+    # recursively find all pretrained in the model config,
+    # and set them None to avoid redundant pretrain steps for testing
+    if hasattr(cfg, 'pretrained'):
```
[the remark this answers is not in the document]
Comment (dreamerlin) [reply tools/test.py:100]: https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/backbones/resnet3d.py#L780

[the remark this answers is not in the document]
Comment (innerlee) [reply tools/test.py:100]: Before the latest commit, pretrained2d was set to None

File: tools/test.py [hunk 6]
```diff
@@ -174,6 +187,9 @@ def main():
                               **cfg.data.get('test_dataloader', {}))
     data_loader = build_dataloader(dataset, **dataloader_setting)
 
+    # remove redundant pretrain steps for testing
+    turn_off_pretrained(cfg.model)
+
```
[the remark this answers is not in the document]
Comment (dreamerlin) [reply tools/test.py:191]: It dose no matter

**Review by kennymckormick — APPROVED** [review]

File: tools/test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dreamerlin) [reply tools/test.py:100]: If `pretrained` is None, `pretrained2d` will affect nothing.

**Review by congee524 — APPROVED** [review]

**Review by congee524 — APPROVED** [review]

**Pull request merged**